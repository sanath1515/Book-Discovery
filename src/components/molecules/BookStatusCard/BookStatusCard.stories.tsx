import React from "react";
import BookStatusCard from "./BookStatusCard";

export default {
  title: "molecules/BookStatusCard",
  component: BookStatusCard,
};

const Template = () => (
  <BookStatusCard
    title="Inorganic Chemistry"
    authorName="Joe Person"
    field="Chemistry"
    image="https://cdn.zeplin.io/5f03041d8c8fc11335546618/assets/B4303A95-C174-4B8B-8F27-D9911CAA7D09.png"
    upCount={16}
    total={100}
    id="1"
  />
);

export const Default = Template.bind({});
