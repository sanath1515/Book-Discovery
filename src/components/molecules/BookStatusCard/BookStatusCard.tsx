import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import Card from "@material-ui/core/Card";
import CardActionArea from "@material-ui/core/CardActionArea";
import CardContent from "@material-ui/core/CardContent";
import CardMedia from "@material-ui/core/CardMedia";
import { CardActions, Typography } from "@material-ui/core";
import { COLORS } from "../../../theme/constants";
import theme from "../../../theme/theme";
import { useDispatch } from "react-redux";
import { useHistory } from "react-router";

const useStyles = makeStyles({
  root: {
    width: "255px",
    height: "108px",
    padding: theme.spacing(2, 0, 2, 1),
    borderRadius: "4px",
    border: "solid 1px",
    borderColor: COLORS.GREY_300,
  },
  image: {
    width: "80px",
    height: "105px",
    margin: theme.spacing(1),
    objectFit: "fill",
  },
  title: {
    width: "152px",
    height: "22px",
    color: COLORS.CHARCOAL_GREY,
    textAlign: "left",
    paddingLeft: theme.spacing(1),
  },
  author: {
    width: "20px",
    height: "18px",
    color: COLORS.ICON_500,
    paddingLeft: theme.spacing(0.5),
  },
  authorName: {
    width: "160px",
    height: "22px",
    paddingTop: theme.spacing(1.5),
    color: COLORS.CHARCOAL_GREY,
    textAlign: "left",
  },
  field: {
    width: "55px",
    height: "18px",
    textTransform: "none",
    color: COLORS.ICON_500,
  },
  fieldName: {
    width: "64px",
    height: "22px",
    textTransform: "none",
    color: COLORS.CHARCOAL_GREY,
    textAlign: "left",
  },
  pages: {
    width: "auto",
    height: "18px",
    textTransform: "none",
    color: COLORS.CHARCOAL_GREY_100,
  },
  actions: {
    padding: theme.spacing(0.5),
    margin: theme.spacing(0, 0.5),
  },
  actionArea: {
    display: "flex",
    width: "auto",
  },
  content: {
    padding: theme.spacing(0),
  },
});
export type BookStatusCardProps = {
  id: string;
  title: string;
  authorName: string;
  field: string;
  image: string;
  upCount?: number;
  total?: number;
};
const BookStatusCard: React.FC<BookStatusCardProps> = ({
  id,
  title,
  field,
  authorName,
  image,
  upCount,
  total,
}) => {
  const classes = useStyles();

  const dispatch = useDispatch();
  const history = useHistory();

  const handleClick = (bookId: string) => {
    dispatch({ type: "SET_SINGLE_VIEW_ID", payload: bookId });
    history.push("/bookdetail");
  };
  return (
    <Card className={classes.root}>
      <CardActionArea className={classes.actionArea}>
        <CardMedia
          className={classes.image}
          component="img"
          image={image}
          onClick={() => handleClick(id)}
          data-testid="card"
        />
        <CardContent className={classes.content}>
          <Typography className={classes.title} variant="subtitle1">
            {title}
          </Typography>
          <CardActions className={classes.actions}>
            <Typography className={classes.author} variant="body2">
              by
            </Typography>
            <Typography className={classes.authorName} variant="body2">
              {authorName}
            </Typography>
          </CardActions>
          <CardActions className={classes.actions}>
            <Typography className={classes.field} variant="overline">
              Category:
            </Typography>
            <Typography className={classes.fieldName} variant="body2">
              {field}
            </Typography>
          </CardActions>
          <CardActions className={classes.actions}>
            <Typography className={classes.pages} variant="overline">
              {upCount} left / {total} pages
            </Typography>
          </CardActions>
        </CardContent>
      </CardActionArea>
    </Card>
  );
};

export default BookStatusCard;
