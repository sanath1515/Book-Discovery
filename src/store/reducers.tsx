export type bookState = {
    searchText:string,
    category:string,
    singleViewId:string
}

export const initialBookState = {
    searchText:"",
    category:"",
    singleViewId:"1"
}

type Action = {type:string,payload:string}

export const booksReducer = (state:bookState = initialBookState,action:Action)=>
{
    switch(action.type)
    {
        case "SET_CATEGORY":{
            return {...state,category:action.payload};
        }
        case "SET_SEARCH_TEXT":{
            return {...state,searchText:action.payload};
        }
        case "SET_SINGLE_VIEW_ID":{
            return {...state,singleViewId:action.payload};
        }
        default:
            return state;
    }
}