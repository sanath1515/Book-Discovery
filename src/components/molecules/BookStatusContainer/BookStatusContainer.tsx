import { Grid, Typography } from "@material-ui/core";
import { makeStyles } from "@material-ui/styles";
import React from "react";
import BookStatusCard from "../BookStatusCard/BookStatusCard";
import ArrowForwardIosIcon from "@material-ui/icons/ArrowForwardIos";
import { COLORS } from "../../../theme/constants";
import theme from "../../../theme/theme";
import { bookProp } from "../../../data/type";

const useStyles = makeStyles({
  root: {
    width: "100%",
    display: "flex",
    flexDirection: "column",
    justifyContent: "space-between",
  },
  heading: {
    width: "475px",
    height: "38px",
    color: COLORS.CHARCOAL_GREY,
    float: "left",
  },
  text: {
    width: "68px",
    height: "24px",
    margin: theme.spacing(3, 1, 1, 0),
    color: COLORS.PRIMARY_COLOR_500,
    float: "right",
  },
  icon: {
    color: COLORS.PRIMARY_COLOR_500,
    margin: theme.spacing(4.5, 0, 1, 0),
    fontSize: "12px",
    float: "right",
  },
  textContainer: {
    width: "100%",
    marginBottom: theme.spacing(2),
  },
  textContainer1: {
    display: "flex",
    justifyContent: "space-between",
  },
});

export type BookStatusContainerProps = {
  data: Array<bookProp>;
};
const BookStatusContainer: React.FC<BookStatusContainerProps> = ({ data }) => {
  const allBookStatusCards = data.map((statusCard) => (
    <BookStatusCard
      image={statusCard.bookImage!}
      authorName={statusCard.author!}
      upCount={statusCard.pagesLeft}
      total={statusCard.totalPages}
      id={statusCard.id!}
      title={statusCard.title!}
      field={statusCard.category!}
    />
  ));

  const classes = useStyles();
  return (
    <div className={classes.root}>
      <Grid className={classes.textContainer}>
        <Typography variant="h3" className={classes.heading}>
          Books you are reading
        </Typography>
        <ArrowForwardIosIcon className={classes.icon} />
        <Typography variant="subtitle2" className={classes.text}>
          See more{" "}
        </Typography>
      </Grid>
      <Grid className={classes.textContainer1}>{allBookStatusCards}</Grid>
    </div>
  );
};

export default BookStatusContainer;
