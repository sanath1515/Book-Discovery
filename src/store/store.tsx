import { createStore } from 'redux';
import { booksReducer } from './reducers';

const store = createStore(booksReducer);

export default store;