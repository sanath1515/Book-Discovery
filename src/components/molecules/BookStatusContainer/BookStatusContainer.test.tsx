import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { BrowserRouter as Router } from "react-router-dom";
import store from "../../../store/store";
import { BookStatusCardProps } from "../BookStatusCard/BookStatusCard";
import BookStatusContainer from "./BookStatusContainer";


const bookData: BookStatusCardProps[] = [
    {
        id: "1",
        title: "Crystal chemistry",
        authorName: "Joe Pearson",
        image:
            "https://cdn.zeplin.io/5f03041d8c8fc11335546618/assets/E3EEB4FA-4DF1-491D-8A51-DC1C1921FB60.png",
        upCount: 0,
        field:"chemistry",
        total: 200
    },
    {
        id: "2",
        title: "Inorganic chemistry",
        authorName: "Joe Pearson",
        image:
            "https://cdn.zeplin.io/5f03041d8c8fc11335546618/assets/03991C69-3988-432F-AD72-535D7FCFFC54.png",
        upCount: 0,
        field:"chemistry",
        total: 200
    },
    {
        id: "3",
        title: "Bioorganic chemistry",
        authorName: "Joe Pearson",
        image:
            "https://cdn.zeplin.io/5f03041d8c8fc11335546618/assets/19118712-5417-4364-9FC6-59FA289CE148.png",
        upCount: 0,
        field:"chemistry",
        total: 200
    },
    {
        id: "4",
        title: "Basic Physics",
        authorName: "Joe Pearson",
        image:
            "https://cdn.zeplin.io/5f03041d8c8fc11335546618/assets/BFB91888-5030-4E35-878F-68EC357D95CE.png",
        upCount: 0,
        field:"chemistry",
        total: 200
    },
];

describe("Should render container correctly", () => {
    const component = (
        <Router>
            <Provider store={store}>
                <BookStatusContainer data={bookData} />
            </Provider>
        </Router>
    );

    it("Match Snapshot", () => {
        const { container } = render(
            <Router>
                <Provider store={store}>
                    <BookStatusContainer data={bookData} />
                </Provider>
            </Router>
        );
        expect(container).toMatchSnapshot();
    });

    test("Should render correctly", () => {
        const { container } = render(
            <Router>
                <Provider store={store}>
                    <BookStatusContainer data={bookData} />
                </Provider>
            </Router>
        );
        expect(container).toBeInTheDocument();
    });
});
