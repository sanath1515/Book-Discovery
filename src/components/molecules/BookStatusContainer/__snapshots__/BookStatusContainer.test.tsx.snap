// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render container correctly Match Snapshot 1`] = `
<div>
  <div
    class="makeStyles-root-1"
  >
    <div
      class="MuiGrid-root makeStyles-textContainer-5"
    >
      <h3
        class="MuiTypography-root makeStyles-heading-2 MuiTypography-h3"
      >
        Books you are reading
      </h3>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root makeStyles-icon-4"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M5.88 4.12L13.76 12l-7.88 7.88L8 22l10-10L8 2z"
        />
      </svg>
      <h6
        class="MuiTypography-root makeStyles-text-3 MuiTypography-subtitle2"
      >
        See more
         
      </h6>
    </div>
    <div
      class="MuiGrid-root makeStyles-textContainer1-6"
    >
      <div
        class="MuiPaper-root MuiCard-root makeStyles-root-7 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <button
          class="MuiButtonBase-root MuiCardActionArea-root makeStyles-actionArea-16"
          tabindex="0"
          type="button"
        >
          <img
            class="MuiCardMedia-root makeStyles-image-8 MuiCardMedia-media MuiCardMedia-img"
            data-testid="card"
          />
          <div
            class="MuiCardContent-root makeStyles-content-17"
          >
            <h6
              class="MuiTypography-root makeStyles-title-9 MuiTypography-subtitle1"
            >
              Crystal chemistry
            </h6>
            <div
              class="MuiCardActions-root makeStyles-actions-15 MuiCardActions-spacing"
            >
              <p
                class="MuiTypography-root makeStyles-author-10 MuiTypography-body2"
              >
                by
              </p>
              <p
                class="MuiTypography-root makeStyles-authorName-11 MuiTypography-body2"
              />
            </div>
            <div
              class="MuiCardActions-root makeStyles-actions-15 MuiCardActions-spacing"
            >
              <span
                class="MuiTypography-root makeStyles-field-12 MuiTypography-overline"
              >
                Category:
              </span>
              <p
                class="MuiTypography-root makeStyles-fieldName-13 MuiTypography-body2"
              />
            </div>
            <div
              class="MuiCardActions-root makeStyles-actions-15 MuiCardActions-spacing"
            >
              <span
                class="MuiTypography-root makeStyles-pages-14 MuiTypography-overline"
              >
                 left / 
                 pages
              </span>
            </div>
          </div>
          <span
            class="MuiCardActionArea-focusHighlight"
          />
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="MuiPaper-root MuiCard-root makeStyles-root-7 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <button
          class="MuiButtonBase-root MuiCardActionArea-root makeStyles-actionArea-16"
          tabindex="0"
          type="button"
        >
          <img
            class="MuiCardMedia-root makeStyles-image-8 MuiCardMedia-media MuiCardMedia-img"
            data-testid="card"
          />
          <div
            class="MuiCardContent-root makeStyles-content-17"
          >
            <h6
              class="MuiTypography-root makeStyles-title-9 MuiTypography-subtitle1"
            >
              Inorganic chemistry
            </h6>
            <div
              class="MuiCardActions-root makeStyles-actions-15 MuiCardActions-spacing"
            >
              <p
                class="MuiTypography-root makeStyles-author-10 MuiTypography-body2"
              >
                by
              </p>
              <p
                class="MuiTypography-root makeStyles-authorName-11 MuiTypography-body2"
              />
            </div>
            <div
              class="MuiCardActions-root makeStyles-actions-15 MuiCardActions-spacing"
            >
              <span
                class="MuiTypography-root makeStyles-field-12 MuiTypography-overline"
              >
                Category:
              </span>
              <p
                class="MuiTypography-root makeStyles-fieldName-13 MuiTypography-body2"
              />
            </div>
            <div
              class="MuiCardActions-root makeStyles-actions-15 MuiCardActions-spacing"
            >
              <span
                class="MuiTypography-root makeStyles-pages-14 MuiTypography-overline"
              >
                 left / 
                 pages
              </span>
            </div>
          </div>
          <span
            class="MuiCardActionArea-focusHighlight"
          />
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="MuiPaper-root MuiCard-root makeStyles-root-7 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <button
          class="MuiButtonBase-root MuiCardActionArea-root makeStyles-actionArea-16"
          tabindex="0"
          type="button"
        >
          <img
            class="MuiCardMedia-root makeStyles-image-8 MuiCardMedia-media MuiCardMedia-img"
            data-testid="card"
          />
          <div
            class="MuiCardContent-root makeStyles-content-17"
          >
            <h6
              class="MuiTypography-root makeStyles-title-9 MuiTypography-subtitle1"
            >
              Bioorganic chemistry
            </h6>
            <div
              class="MuiCardActions-root makeStyles-actions-15 MuiCardActions-spacing"
            >
              <p
                class="MuiTypography-root makeStyles-author-10 MuiTypography-body2"
              >
                by
              </p>
              <p
                class="MuiTypography-root makeStyles-authorName-11 MuiTypography-body2"
              />
            </div>
            <div
              class="MuiCardActions-root makeStyles-actions-15 MuiCardActions-spacing"
            >
              <span
                class="MuiTypography-root makeStyles-field-12 MuiTypography-overline"
              >
                Category:
              </span>
              <p
                class="MuiTypography-root makeStyles-fieldName-13 MuiTypography-body2"
              />
            </div>
            <div
              class="MuiCardActions-root makeStyles-actions-15 MuiCardActions-spacing"
            >
              <span
                class="MuiTypography-root makeStyles-pages-14 MuiTypography-overline"
              >
                 left / 
                 pages
              </span>
            </div>
          </div>
          <span
            class="MuiCardActionArea-focusHighlight"
          />
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="MuiPaper-root MuiCard-root makeStyles-root-7 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <button
          class="MuiButtonBase-root MuiCardActionArea-root makeStyles-actionArea-16"
          tabindex="0"
          type="button"
        >
          <img
            class="MuiCardMedia-root makeStyles-image-8 MuiCardMedia-media MuiCardMedia-img"
            data-testid="card"
          />
          <div
            class="MuiCardContent-root makeStyles-content-17"
          >
            <h6
              class="MuiTypography-root makeStyles-title-9 MuiTypography-subtitle1"
            >
              Basic Physics
            </h6>
            <div
              class="MuiCardActions-root makeStyles-actions-15 MuiCardActions-spacing"
            >
              <p
                class="MuiTypography-root makeStyles-author-10 MuiTypography-body2"
              >
                by
              </p>
              <p
                class="MuiTypography-root makeStyles-authorName-11 MuiTypography-body2"
              />
            </div>
            <div
              class="MuiCardActions-root makeStyles-actions-15 MuiCardActions-spacing"
            >
              <span
                class="MuiTypography-root makeStyles-field-12 MuiTypography-overline"
              >
                Category:
              </span>
              <p
                class="MuiTypography-root makeStyles-fieldName-13 MuiTypography-body2"
              />
            </div>
            <div
              class="MuiCardActions-root makeStyles-actions-15 MuiCardActions-spacing"
            >
              <span
                class="MuiTypography-root makeStyles-pages-14 MuiTypography-overline"
              >
                 left / 
                 pages
              </span>
            </div>
          </div>
          <span
            class="MuiCardActionArea-focusHighlight"
          />
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
